backend/src/user/user.controller.ts

```diff
@@ -68,4 +69,19 @@ export class UserController {
     const { name } = user;
     return { name };
   }
+
+  @UseGuards(JwtAuthGuard)
+  @Post(":userId(\\d+)")
+  async updatePassword(
+    @Request() req: ExpressRequest,
+    @Param("userId", ParseIntPipe) userId: number,
+    @Body() data: UpdatePasswordReq,
+  ): Promise<void> {
+    if (req.user?.userId != userId) {
+      throw new ForbiddenException();
+    }
+    const { newPassword } = data;
+    const { oldPassword } = data;
+    await this.userService.updateUserPassword(userId, newPassword, oldPassword);
```

Comment:
I think the password arguments are reversed. 
```suggestion
    const { oldPassword, newPassword } = data;
    await this.userService.updateUserPassword(userId, oldPassword, newPassword);
```